packages/examples/cvat/recording-oracle/src/handlers/process_intermediate_results.py

```diff
@@ -980,17 +405,126 @@ def process_intermediate_results(  # noqa: PLR0912
+                    if honeypot in job_honeypots
+                ]
+                sorted_task_frame_names = validation_result.task_id_to_sequence_of_frame_names[
+                    cvat_task_id
+                ]
+
+                for val_frame in validation_frames:
+                    val_frame_name = sorted_task_frame_names[val_frame]
+                    val_frame_stats = gt_stats[val_frame_name]
+                    if (
+                        val_frame_stats.failed_attempts >= Config.validation.gt_ban_threshold
+                        and not val_frame_stats.accepted_attempts
+                    ):
+                        cvat_task_id_to_failed_val_frames.setdefault(cvat_task_id, set()).add(
+                            val_frame
+                        )
+
+        for cvat_task_id, val_frame_ids in cvat_task_id_to_failed_val_frames.items():
+            task_validation_layout = validation_result.task_id_to_val_layout[cvat_task_id]
+            intersection = val_frame_ids & set(task_validation_layout.disabled_frames)
+
+            if intersection:
+                logger.error(
+                    "Logical error occurred while disabling validation frames "
+                    f"for the task({task_id}). Frames {intersection} "
+                    "are already disabled."
+                )
+
+            updated_disable_frames = task_validation_layout.disabled_frames + list(val_frame_ids)
+            non_disabled_frames = list(
+                set(task_validation_layout.validation_frames) - set(updated_disable_frames)
+            )
+
+            updated_task_honeypot_real_frames = task_validation_layout.honeypot_real_frames.copy()
+
+            # validation frames may be repeated
+            task_honeypot_real_frames_index: dict[int, list[int]] = {}
+            for idx, f in enumerate(updated_task_honeypot_real_frames):
+                task_honeypot_real_frames_index.setdefault(f, []).append(idx)
+
+            rejected_jobs_for_task = [
+                j
+                for j in unchecked_jobs_meta.jobs
+                if j.job_id in rejected_job_ids and j.task_id == cvat_task_id
+            ]
+
+            for job in rejected_jobs_for_task:
+                job_frame_range = job.job_frame_range
+                honeypots_mapping = validation_result.task_id_to_honeypots_mapping[cvat_task_id]
+                job_honeypots = set(honeypots_mapping.keys()) & set(job_frame_range)
+
+                validation_frames_to_replace = [
+                    honeypots_mapping[honeypot]
+                    for honeypot in job_honeypots
+                    if honeypots_mapping[honeypot] in val_frame_ids
+                ]
+
+                # choose new unique validation frames for the job
+                available_validation_frames = list(
+                    set(non_disabled_frames) - set(validation_frames_to_replace)
+                )
+                rng = np.random.default_rng()
```

Comment:
Please use a specific known generator. We don't know when the default might change. MT19937 would be ok.